### PR TITLE
Points docs to last stable release

### LINKED ISSUE #157: latest docs should point to the latest non-experimental release, not master
Currently in the https://www.fastify.io/ top banner, we point to docs of master.
With the upcoming v3 release, this would soon be experimental/alpha. So, we should point the docs to the latest stable version.

### PR DESCRIPTION
**Pull request by lmammino**
Closes #157 

This PR changes the current behaviour to point to the `latest` documentation and provides some high-level refactoring of the build process code.

As discussed in #157, the new behaviour is the following:

- `latest` doesn't track to `master` anymore, the 2 *tags* are now independent
- `latest` will point to the most recent version that does not include annotation (as for instance *-alpha.1*)
- `master` tracks to actual `master` (this is unchanged)

I created a fake `v3.0.0-alpha` to test these assumptions. You can see a preview in the demo below.

Please help me to test this thoroughly before we ship any change :) 

DEMO: https://latest-doc-to-last-stable-release.surge.sh/

### CODE REVIEW

File: src/scripts/downloadReleases.js [hunk 1]
```diff
@@ -1,87 +1,160 @@
...
+      {
+        name: 'v3.0.0-alpha',
+        zipball_url: 'https://api.github.com/repos/fastify/fastify/zipball/v1.13.3'
+      },
+      {
+        name: 'v3.0.0-alpha.2',
+        zipball_url: 'https://api.github.com/repos/fastify/fastify/zipball/v1.13.3'
+      }
+    ])
+    // TODO: ----------------------------------------------------------------------- end of debug
```
Comment (lmammino) [inline src/scripts/downloadReleases.js]: **Note**: This is used to simulate an upcoming alpha release, it will be removed before the merge

Comment (lmammino) [inline src/scripts/downloadReleases.js]: **Note**: This is used to simulate two upcoming alpha release, it will be removed before the merge

**Review by mcollina — APPROVED** [review]
lgtm

**Review by jsumners — APPROVED** [review]

**Review by Eomm — APPROVED** [review]
LGTM!

Only a question:
when you switch version in a page (like decorators for ex), the user is sent to the TOC, could we maintain the path changing only the version-path-param?

**Comment by lmammino** [comment]
> Only a question:
when you switch version in a page (like decorators for ex), the user is sent to the TOC, could we maintain the path changing only the version-path-param?

In theory yes, in practice that page might not exist in another version. Not sure right now what would be a better way to handle this to be honest without having to map all the possible pages in advance.

**Comment by lmammino** [comment]
@Eomm, actually, on a second thought, there might be a way to achieve that. The question is,
when a page does not exist in a given version, what do we do?

- Option 1: we don't show the version itself in the dropdown
- Option 2: we show the version but we point it to the Table of Contents for that version

Any preference @Eomm, @mcollina, @jsumners ?

**Comment by jsumners** [comment]
I'd make it a follow-up PR.

**Comment by lmammino** [comment]
@jsumners makes sense. Then I will merge this and I will create a dedicated issue to discuss this other potential improvement :)

**Comment by lmammino** [comment]
Merged, released and open a dedicated issue (#161) for the problem raised by @Eomm.

Thanks a lot everyone for the quick turnout on this! 🚀

**Pull request merged**